calico_kubernetes/calico_kubernetes.py

```diff
@@ -39,8 +40,9 @@
 from pycalico.ipam import IPAMClient
 from pycalico.util import generate_cali_interface_name, get_host_ips
 
-logger = logging.getLogger(__name__)
 pycalico_logger = logging.getLogger(pycalico.__name__)
+base_logger = logging.getLogger(__name__)
+logger = logging.LoggerAdapter(base_logger, {'docker_id': "unknown"})
```

Comment:
Rather than write "[unknown]" in all the logs that occur when we don't have an ID, can't we just change the logger's formatter to add the `docker_id` variable at the point that we learn the id (i.e. line 751)? That way we don't print useless log characters in the no-ID case.